project/FrgaalJavaCompiler.scala

```diff
@@ -63,6 +64,58 @@ object FrgaalJavaCompiler {
     val sources = sources0 map {
       case x: PathBasedFile => x.toPath.toAbsolutePath.toString
     }
+    val out = output.getSingleOutputAsPath().get()
+    val shared = sources0.fold(out)((a, b) => {
+      var ap = a match {
+        case p: PathBasedFile => p.toPath
+        case p: Path => p
+      }
+      val bp = b match {
+        case p: PathBasedFile => p.toPath
+        case p: Path => p
+      }
+
+      var i = 0
+      while (i < Math.min(ap.getNameCount(), bp.getNameCount()) && ap.getName(i) == bp.getName(i)) {
+        i += 1;
+      }
+
+      while (ap.getNameCount() > i) {
+        ap = ap.getParent()
+      }
+      ap
+    }).asInstanceOf[Path];
+
+    if (shared.toFile().exists()) {
+      val ensoMarker = new File(shared.toFile(), ".enso-sources")
+      val ensoConfig = new File(shared.toFile(), ".enso-sources-" + out.getFileName().toString())
+      val p = new java.util.Properties()
+
+      def storeArray(name: String, values : Seq[String]) = {
+        var at = 0
+        while (values.length > at) {
+          p.setProperty(name + "." + at, values(at))
+          at += 1
+        }
```

Comment:
_the Scala way_ would be to rather
```suggestion
        values.zipWithIndex.foreach { case (value, idx) => p.setProperty(s"$name.$idx", value) }
```